### PR TITLE
Add checked float-to-int helper function

### PR DESCRIPTION
**Pull request by eduardosm**
As discussed in https://github.com/rust-lang/miri/pull/2989#discussion_r1287252367

### CODE REVIEW

File: src/shims/intrinsics/mod.rs [hunk 1]
```diff
@@ -393,47 +393,10 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         F: Float + Into<Scalar<Provenance>>,
     {
```
Comment (RalfJung) [inline src/shims/intrinsics/mod.rs:394]: This is not worth its own function any more (and having 2 functions with the same name is anyway potentially confusing).

Comment (eduardosm) [reply src/shims/intrinsics/mod.rs]: Inlined the function

File: src/shims/intrinsics/simd.rs [hunk 2]
```diff
@@ -420,7 +420,6 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     }
                 }
             }
-            #[rustfmt::skip]
```
Comment (eduardosm) [inline src/shims/intrinsics/simd.rs:423]: I wasn't sure about the purpose of this

Comment (RalfJung) [reply src/shims/intrinsics/simd.rs:423]: It avoided a bunch of reformatting that I considered undesirable. But now the branches have fairly different sizes anyway so it seems okay.

**Comment by RalfJung** [comment]
Thanks!
@bors r+

**Pull request merged**